src/cross_toml.rs

```diff
@@ -69,12 +99,60 @@ impl CrossToml {
 
     /// Returns the `target.{}.image` part of `Cross.toml`
     pub fn image(&self, target: &Target) -> Option<String> {
-        self.get_string(target, |t| &t.image)
+        self.get_string(target, |_| None, |t| t.image.as_ref())
+    }
+
+    /// Returns the `{}.dockerfile` or `{}.dockerfile.file` part of `Cross.toml`
+    pub fn dockerfile(&self, target: &Target) -> Option<String> {
+        self.get_string(
+            target,
+            |b| b.dockerfile.as_ref().map(|c| &c.file),
+            |t| t.dockerfile.as_ref().map(|c| &c.file),
+        )
+    }
+
+    /// Returns the `target.{}.dockerfile.context` part of `Cross.toml`
+    pub fn dockerfile_context(&self, target: &Target) -> Option<String> {
+        self.get_string(
+            target,
+            |b| b.dockerfile.as_ref().and_then(|c| c.context.as_ref()),
+            |t| t.dockerfile.as_ref().and_then(|c| c.context.as_ref()),
+        )
+    }
+
+    /// Returns the `target.{}.dockerfile.build_args` part of `Cross.toml`
+    pub fn dockerfile_build_args(&self, target: &Target) -> Option<HashMap<String, String>> {
```

Comment:
I wonder if this could use a helper? For 2 optional Vecs to merge them?

`merge_vec(build, target)` where we have:

```rust
fn merge_vec(x: Option<String>, y: Option<String>) -> Option<String>;
```